### PR TITLE
Add doctor check for Python 'six' module

### PR DESCRIPTION
**Pull request by cbracken**
Required as part of Xcode lldb module. In all likelihood, if we
encounter this situation, the developer is using a custom Python install
(e.g., via MacPorts or Homebrew).

### CODE REVIEW

**Comment by cbracken** [comment]
Fixes a potential gotcha encountered in #7767.

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart [hunk 1]
```diff
@@ -90,6 +93,17 @@ class IOSWorkflow extends DoctorValidator implements Workflow {
...
     }
 
+    // Python dependencies installed
+    if (hasPythonSixModule) {
+      pythonStatus = ValidationType.installed;
+    } else {
+      pythonStatus = ValidationType.missing;
+      messages.add(new ValidationMessage.error(
+        'Python installation missing module "six".\n'
+        'Install via \'sudo easy_install six\'.'
```
Comment (xster) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: pip? I think easy_install is outdated

**Review by xster — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/ios/ios_workflow.dart (hunk 1 above)
Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: `pip` doesn't appear to be installed by default, though I wouldn't be surprised if most custom installs list it as a dependency.

Comment (xster) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: Ah good point. Should we just say Install via pip install six or sudo easy_install six? Just so we don't default to the non-recommended way?

Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: SGTM. I'll tweak the message. Another alternative is to mention that they may be using a custom install and suggest that they can fix this by using the default system version (without being to pushy about it).

Comment (cbracken) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: Updated to suggest pip or easy_install (in that order). I couldn't think of a concise way of suggesting that they switch back to the system default and TBH, it'd be nice not to force people away from their preferences, so leaving at that :)

Comment (xster) [inline packages/flutter_tools/lib/src/ios/ios_workflow.dart]: Ya, it's hard to make that assumption. They can be missing the module for all sorts of weird reasons. 

Some soft-ish language like you had is probably fine. Like your thing is missing, we don't presume to know why but if you wanted to install it, you can do it like this:

**Pull request merged**